### PR TITLE
add contained subscriber for reading

### PR DESCRIPTION
**Pull request by eclendenning-va**

### CODE REVIEW

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/coverage/InsuranceBufferToR4CoverageTransformer.java [hunk 1]
```diff
@@ -332,6 +338,85 @@ private CodeableConcept relationship(String ptRelationshipHipaa) {
...
+                                    .build()
+                                    .asList())
+                            .build())
+                    .system("http://hl7.org/fhir/sid/us-ssn")
+                    .value(ssn)
+                    .assigner(
+                        Reference.builder().display("United States Social Security Number").build())
+                    .build()
+                    .asList())
+        .orElse(emptyList());
```
Comment (jhulbert-va) [inline vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/coverage/InsuranceBufferToR4CoverageTransformer.java:417]: This is gonna end with us returning an empty list for this field when it's empty.

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/coverage/InsuranceBufferToR4CoverageTransformer.java [hunk 2]
```diff
@@ -277,6 +256,33 @@ private Identifier insurancePlanIdentifier(
     return InsurancePlan.Plan.builder().type(typeOfPlan).build().asList();
   }
 
+  private List<Address> payorAddress(FilemanEntry entry) {
```
[the remark this answers is not in the document]
Comment (eclendenning-va) [reply vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/coverage/InsuranceBufferToR4CoverageTransformer.java:259]: I tried, but they both require separate fields. this seemed cleaner.

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/coverage/InsuranceBufferToR4CoverageTransformer.java [hunk 3]
```diff
@@ -186,6 +160,11 @@ private Reference beneficiary(@NonNull String patientIcn, Optional<String> maybe
                 .containedResource(toOrganization(entry))
                 .applyReferenceId(
                     (o, id) -> o.payor(Reference.builder().reference(id).build().asList()))
+                .build(),
+            ContainableResource.<Coverage, RelatedPerson>builder()
+                .containedResource(toRelatedPerson(entry))
+                .applyReferenceId(
+                    (c, id) -> c.subscriber(Reference.builder().reference(id).build()))
```
[the remark this answers is not in the document]
Comment (eclendenning-va) [reply vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/coverage/InsuranceBufferToR4CoverageTransformer.java:167]: `RelatedPerson.class.getSimpleName()` ?

**Review by jhulbert-va — APPROVED** [review]
Looks good to me.

**Pull request merged**